providers/oidc.go

```diff
@@ -279,27 +284,32 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 	return claims, nil
 }
 
-func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) []string {
+func (p *OIDCProvider) extractGroupsFromRawClaims(rawClaims map[string]interface{}) ([]string, error) {
 	groups := []string{}
 
 	rawGroups, ok := rawClaims[p.GroupsClaim].([]interface{})
 	if rawGroups != nil && ok {
 		for _, rawGroup := range rawGroups {
 			group, ok := rawGroup.(string)
-			if ok {
-				groups = append(groups, group)
+			if !ok {
+				jsonGroup, err := json.Marshal(rawGroup)
+				if err != nil {
+					return []string{}, err
```

Comment:
Do we really want to error out and discard all previously valid groups we might've extracted? Or log the failure case and not add it into the session's group list.

Shouldn't be a security vulnerability if people are relying on this for group based authorization, because if the group fails to extract, we don't add it to the group list.

Reply:
I decided to log the failure case.